### PR TITLE
port buffered columns to 1.2

### PR DESCRIPTION
**Pull request by Dronplane**

### CODE REVIEW

File: core/utils/file_utils.hpp [hunk 1]
```diff
@@ -128,7 +130,9 @@ bool mtime(time_t& result, const file_path_t file) noexcept;
 // -----------------------------------------------------------------------------
 // --SECTION--                                                         open file
 // -----------------------------------------------------------------------------
-enum class OpenMode { Read, Write };
+enum class OpenMode : uint16_t { Invalid = 0, Read = 1, Write = 2, Direct = 4 };
+
+ENABLE_BITMASK_ENUM(OpenMode);
```
Comment (MBkkt) [inline core/utils/file_utils.hpp:135]: It's not really valid, only operator& valid

File: core/utils/file_utils.cpp [hunk 2]
```diff
@@ -769,16 +778,28 @@ handle_t open(const file_path_t path, OpenMode mode, int advice) noexcept {
...
     IR_FRMT_ERROR(
       "Failed to open file, error: %d, path: " IR_FILEPATH_SPECIFIER, errno,
       path);
     return handle_t(nullptr);
   }
+#ifdef __APPLE__
+  // Implement Alignment and File Access Requirements
+  // if ((mode & OpenMode::Direct) == OpenMode::Direct) {
+  //  fcntl(fd, F_NOCACHE, 1);
+  //}
```
Comment (MBkkt) [inline core/utils/file_utils.cpp:801]:
```suggestion
  // }
```

File: core/utils/file_utils.cpp [hunk 3]
```diff
@@ -769,16 +778,28 @@ handle_t open(const file_path_t path, OpenMode mode, int advice) noexcept {
...
-           (OpenMode::Read == mode ? O_RDONLY : (O_CREAT | O_TRUNC | O_WRONLY)),
-           S_IRUSR | S_IWUSR);
+  auto posix_mode = (((OpenMode::Read & mode) == OpenMode::Read)
+                       ? O_RDONLY
+                       : (O_CREAT | O_TRUNC | O_WRONLY));
+#ifndef __APPLE__
+  // Implement Alignment and File Access Requirements
+  // if ((mode & OpenMode::Direct) == OpenMode::Direct) {
+  //  posix_mode |= O_DIRECT;
+  //}
```
Comment (MBkkt) [inline core/utils/file_utils.cpp:788]:
```suggestion
  // }
```

File: core/store/store_utils.hpp [hunk 4]
```diff
@@ -399,38 +398,86 @@ class bytes_ref_input : public index_input {
...
     return irs::read<uint64_t>(pos_);
   }
 
-  virtual uint64_t read_vlong() noexcept override final {
+  uint64_t read_vlong() noexcept override final {
     return irs::vread<uint64_t>(pos_);
   }
 
-  virtual uint32_t read_vint() noexcept override final {
+  uint32_t read_vint() noexcept override final {
```
Comment (MBkkt) [inline core/store/store_utils.hpp:423]: NIT
remove override

**Review by MBkkt — CHANGES_REQUESTED** [review]

**Review by gnusi — APPROVED** [review]

**Review by MBkkt — APPROVED** [review]

**Pull request merged**